### PR TITLE
[sw] Avoid GNU Extensions for C

### PR DESCRIPTION
**Pull request by lenary**
We want to keep the C that we are writing as close to standard C as
possible. This means switching off `-std=gnu99` in favour of `-std=c99`,
which should lead to better compiler warnings and errors.

Things needed to make this work:
- the variadic logging functions needed to be updated, as there's an issue in the C standard with `__VA_ARGS__` expanding to no arguments.
- a define is needed for mpsse to ensure `usleep` is available. This is added as a patch and a re-vendoring.
- an `asm` macro (expands to `__asm__`), as noted in the Assembly Style Guide

This commit does not update opentitan from C99 to C11, as that change has not yet been agreed.

### CODE REVIEW

**Comment by mcy** [comment]
Shouldn't this also include a blanket ban of __attribute__? We use it in a handful of places when we should probably be hiding it behind macros.

**Comment by lenary** [comment]
`__attribute__` will do the right thing with `-std=c11`, which isn't true for `asm` or the variadic macros (because `__attribute__` is a valid reserved identifier).

Given there's debate on the C11 style guide PR around whether to hide attributes behind a macro, I think sorting out `__attribute__` usage is something we can do one we've settled those issues, in a separate PR.

File: sw/device/lib/log.h [hunk 1]
```diff
@@ -141,7 +141,7 @@
  */
 #ifndef PRINT_LOG
 #warning PRINT_LOG undefined: it will result in vacuous invocation
-#define PRINT_LOG(log_header, fmt, ...)
+#define PRINT_LOG(log_header, ...)
```
Comment (tjaychen) [inline sw/device/lib/log.h]: does removing fmt here change the behavior of the logging function..?

Comment (gkelly) [reply sw/device/lib/log.h]: It was intentional to ensure that zero-length varargs don't get built.  The varargs now have the format string in them and this gets used as the first argument.

Comment (lenary) [reply sw/device/lib/log.h]: @tjaychen do you have your own implementations of `PRINT_LOG` or the other similar macros in this file? If so, their signature may need to be updated, but the functionality should not be different.

**Pull request merged**